src/main/kotlin/io.javalin/Javalin.kt

```diff
@@ -0,0 +1,304 @@
+
+class Javalin {
+
+    private val log = LoggerFactory.getLogger(Javalin::class.java)
+
+    companion object {
+        var DEFAULT_PORT = 7000
+        @JvmStatic fun create(): Javalin {
+            return Javalin()
+        }
+    }
+
+    private var port = DEFAULT_PORT
+
+    private var ipAddress = "0.0.0.0"
+
+    private var embeddedServer: EmbeddedServer? = null
+    private var embeddedServerFactory: EmbeddedServerFactory = EmbeddedJettyFactory()
+
+    private var staticFileDirectory: String? = null
+    private var pathMatcher = PathMatcher()
+    private var exceptionMapper = ExceptionMapper()
+    private var errorMapper = ErrorMapper()
+
+    fun clearInternalMappers() {
+        pathMatcher.clear();
+        exceptionMapper.clear();
+        errorMapper.clear();
+    }
+
+    private val eventManager = EventManager()
+
+    private var startLatch = CountDownLatch(1)
+    private var stopLatch = CountDownLatch(1)
+
+    private var accessManager = AccessManager { handler: Handler, req: Request, res: Response, roles: List<Role> ->
+        throw IllegalStateException("No access manager configured. Add an access manager using 'accessManager()'")
+    }
+
+    fun accessManager(accessManager: AccessManager): Javalin {
+        this.accessManager = accessManager
+        return this
+    }
+
+    private var started = false
+
+    @Synchronized fun start(): Javalin {
+        if (!started) {
+            log.info(Util.javalinBanner())
+            Util.printHelpfulMessageIfLoggerIsMissing()
+            Thread {
+                eventManager.fireEvent(Event.Type.SERVER_STARTING, this)
+                try {
+                    embeddedServer = embeddedServerFactory.create(pathMatcher, exceptionMapper, errorMapper, staticFileDirectory)
+                    port = embeddedServer!!.start(ipAddress, port)
+                } catch (e: Exception) {
+                    log.error("Failed to start Javalin")
+                    eventManager.fireEvent(Event.Type.SERVER_START_FAILED, this)
+                }
+
+                eventManager.fireEvent(Event.Type.SERVER_STARTED, this)
+                try {
+                    startLatch.countDown()
+                    embeddedServer!!.join()
+                } catch (e: InterruptedException) {
+                    log.error("Server startup interrupted", e)
+                    Thread.currentThread().interrupt()
+                }
+            }.start()
+            started = true
+        }
+        return this
+    }
+
+    fun awaitInitialization(): Javalin {
+        if (!started) {
+            throw IllegalStateException("Server hasn't been started. Call start() before calling this method.")
+        }
+        try {
+            startLatch.await()
+        } catch (e: InterruptedException) {
+            log.info("awaitInitialization was interrupted")
+            Thread.currentThread().interrupt()
+        }
+
+        return this
+    }
+
+    @Synchronized fun stop(): Javalin {
```

Comment:
probably outside the scope of a review of porting this to kotlin, but i'm a little confused by this - is everything reset so that it can be reused?